nullaway/src/test/java/com/uber/nullaway/EnsuresNonNullIfTests.java

```diff
@@ -0,0 +1,801 @@
+        .doTest();
+  }
+
+  @Test
+  public void setTrueIfNonNullToFalse() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class Foo {",
+            "  @Nullable Item nullableItem;",
+            "  @EnsuresNonNullIf(value=\"nullableItem\", result=false)",
+            "  public boolean doesNotHaveNullableItem() {",
+            "    return nullableItem == null;",
+            "  }",
+            "  public int runOk() {",
+            "    if(doesNotHaveNullableItem()) {",
+            "      return 1;",
+            "    }",
+            "    nullableItem.call();",
+            "    return 0;",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Item.java", "package com.uber;", "class Item {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  @Test
+  public void setTrueIfNonNullToFalseMultipleElements() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class Foo {",
+            "  @Nullable Item nullableItem;",
+            "  @Nullable Item nullableItem2;",
+            "  @EnsuresNonNullIf(value={\"nullableItem\",\"nullableItem2\"}, result=false)",
+            "  public boolean doesNotHaveNullableItem() {",
+            "    return nullableItem == null && nullableItem2 == null;",
+            "  }",
+            "  public int runOk() {",
+            "    if(doesNotHaveNullableItem()) {",
+            "      return 1;",
+            "    }",
+            "    nullableItem.call();",
+            "    nullableItem2.call();",
+            "    return 0;",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Item.java", "package com.uber;", "class Item {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  @Test
+  public void setTrueIfNonNullToFalseMultipleElements_deferenceFound() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class Foo {",
+            "  @Nullable Item nullableItem;",
+            "  @Nullable Item nullableItem2;",
+            "  @EnsuresNonNullIf(value={\"nullableItem\",\"nullableItem2\"}, result=false)",
+            "  public boolean doesNotHaveNullableItem() {",
+            "    // BUG: Diagnostic contains: Method is annotated with @EnsuresNonNullIf but does not ensure fields [nullableItem, nullableItem2]",
+            "    return nullableItem == null || nullableItem2 == null;",
+            "  }",
```

Comment:
This annotation _should_ pass validation.  If the function returns `false`, we know that neither of the fields can be `null`, i.e., both are non-null.

Reply:
Are you a person or a compiler? Great catch!